src/views/TradingCompetition/components/Ribbon/index.tsx

```diff
@@ -0,0 +1,108 @@
+import React from 'react'
+import styled from 'styled-components'
+import { Flex, LaurelLeftIcon, LaurelRightIcon } from '@pancakeswap-libs/uikit'
+import { RibbonProps } from '../../types'
+import {
+  RibbonDownMidExpanding,
+  RibbonUpMidExpanding,
+  RibbonDownRightSide,
+  RibbonUpRightSide,
+  RibbonDownLeftSide,
+  RibbonUpLeftSide,
+} from './RibbonStyles'
+import { Heading2Text, VisuallyHiddenHeading2Text } from '../CompetitionHeadingText'
+
+const Wrapper = styled(Flex)<{ ribbonDirection?: 'up' | 'down' }>`
+  position: relative;
+  display: inline-flex;
+  align-items: ${({ ribbonDirection }) => (ribbonDirection === 'up' ? 'flex-end' : 'flex-start')};
+  justify-content: center;
+
+  svg {
+    z-index: 200;
+  }
+`
+
+const TextWrapper = styled(Flex)`
+  display: flex;
+  align-items: center;
+  position: absolute;
+  height: 46px;
+  background-color: #7645d9;
+`
+
+const LaurelWrapper = styled.div<{ dir?: 'l' | 'r' }>`
+  display: flex;
+  align-items: center;
+  justify-content: center;
+  position: absolute;
+  z-index: 2;
+  top: 50%;
+  transform: translate(0, -50%);
+  ${({ dir }) =>
+    dir === 'l'
```

Comment:
wtf is this formatting 😂

Reply:
OH this? I added this to make your eyes bleed.

Reply:
I actually think this was how linting on my old work laptop was setup. Absolutely brutal eh.